alicloud/resource_alicloud_snapshot_test.go

```diff
@@ -0,0 +1,291 @@
+package alicloud
+
+import (
+	"fmt"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+	"log"
+	"strings"
+	"testing"
+)
+
+func init() {
+	resource.AddTestSweepers("alicloud_snapshot", &resource.Sweeper{
+		Name: "alicloud_snapshot",
+		F:    testSweepSnapshots,
+	})
+}
+
+func testSweepSnapshots(region string) error {
+	rawClient, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting Alicloud client: %s", err)
+	}
+	client := rawClient.(*connectivity.AliyunClient)
+
+	prefixes := []string{
+		"tf-testAcc",
+		"tf_testAcc",
+	}
+
+	var snapshots []ecs.Snapshot
+	req := ecs.CreateDescribeSnapshotsRequest()
+	req.RegionId = client.RegionId
+	req.PageSize = requests.NewInteger(PageSizeLarge)
+	req.PageNumber = requests.NewInteger(1)
+	for {
+		raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+			return ecsClient.DescribeSnapshots(req)
+		})
+		if err != nil {
+			return fmt.Errorf("Error retrieving snapshots: %s", err)
+		}
+		resp, _ := raw.(*ecs.DescribeSnapshotsResponse)
+		if resp == nil || len(resp.Snapshots.Snapshot) < 1 {
+			break
+		}
+		snapshots = append(snapshots, resp.Snapshots.Snapshot...)
+
+		if len(resp.Snapshots.Snapshot) < PageSizeLarge {
+			break
+		}
+
+		if page, err := getNextpageNumber(req.PageNumber); err != nil {
+			return err
+		} else {
+			req.PageNumber = page
+		}
+	}
+
+	for _, v := range snapshots {
+		name := v.SnapshotName
+		id := v.SnapshotId
+		skip := true
+		for _, prefix := range prefixes {
+			if strings.HasPrefix(strings.ToLower(name), strings.ToLower(prefix)) {
+				skip = false
+				break
+			}
+		}
+		if skip {
+			log.Printf("[INFO] Skipping snapshot: %s (%s)", name, id)
+			continue
+		}
+		log.Printf("[INFO] Deleting snapshot: %s (%s)", name, id)
+		req := ecs.CreateDeleteSnapshotRequest()
+		req.SnapshotId = id
+		_, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+			return ecsClient.DeleteSnapshot(req)
+		})
+		if err != nil {
+			log.Printf("[ERROR] Failed to delete snapshot(%s (%s)): %s", name, id, err)
+		}
+	}
```

Comment:
If the delete api is not sync, please set several waiting seconds after deleting.